src/model.rs

```diff
@@ -787,88 +792,79 @@ impl Collections {
     ///  VJ:2 (Mon-Fri)                                       12:00-13:00       14:00-15:00
     ///  VJ:3 (Sat-Sun)                                       12:30-13:30       14:30-15:30
     ///
-    /// Example 4 might be a valid use case of stay-in but is not handled in the
-    /// current implementation since the validity patterns are different; this
-    /// is undefined behavior (most likely, some stay-in will be forbidden by
-    /// the default rule).
+    /// Example 4 is a valid use case of stay-in
+    /// The pickup/dropoff will be possible between VJ:1 and VJ:2/VJ:3
     pub fn enhance_pickup_dropoff(&mut self) {
-        let mut vj_idxs: Vec<Idx<VehicleJourney>> = self
-            .vehicle_journeys
+        let mut allowed_last_pick_up_vj = HashSet::<_>::new();
+        let mut allowed_first_drop_off_vj = HashSet::<_>::new();
+
+        let can_chain_without_overlap = |prev_vj: &VehicleJourney, next_vj: &VehicleJourney| {
+            let last_stop = &prev_vj.stop_times.last();
+            let first_stop = &next_vj.stop_times.first();
+            match (last_stop, first_stop) {
+                (Some(last_stop), Some(first_stop)) => {
```

Comment:
hum the problem is that we cannot put a comment on the specific ìf` to explain that it handles the example 1 :/
I'm not sure it will help the understanding (I did not understand this ìf` guard immediatly in the previous implementation)

Reply:
To me, this is the level of nesting in this whole function that is making it hard to read and understand. As for the comment, it might be ok and become something like this.

> We can discard when the stop points are identicals (see Example 1 above) or when there is no stop point

Because basically, if there is only 1 stop point, then they're first and last are going to be equal.